core/arch/arm/arm.mk

```diff
@@ -166,7 +168,17 @@ ta_arm32-platform-cflags += $(arm32-platform-cflags)
 ta_arm32-platform-cflags += $(platform-cflags-optimization)
 ta_arm32-platform-cflags += $(platform-cflags-debug-info)
 ta_arm32-platform-cflags += -fpic
-ta_arm32-platform-cflags += $(arm32-platform-cflags-generic)
+
+# Thumb mode doesn't support function graph tracing due to missing
+# frame pointer support required to trace function call chain. So
+# rather compile in ARM mode if function tracing is enabled.
+ifeq ($(CFG_TA_FTRACE_SUPPORT),y)
+ta_arm32-platform-cflags += $(arm32-platform-cflags-generic-arm) \
+			-fno-omit-frame-pointer -mapcs
```

Comment:
`-fno-omit-frame-pointer -mapcs` flags adds some overhead so I kept it under ftrace. But since `arm32-platform-cflags-generic-arm` isn't being used elsewhere, so I could move the flags.